lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImpl.java

```diff
@@ -26,56 +27,52 @@
  * Implementation of the {@link FusedLocationProviderApi}.
  */
 public class FusedLocationProviderApiImpl
-    implements FusedLocationProviderApi {
+    implements FusedLocationProviderApi, EventCallbacks, ServiceConnection {
 
   private Context context;
   private FusedLocationProviderService service;
   private FusedLocationServiceConnectionManager serviceConnectionManager;
+  private boolean isBound;
 
-  private FusedLocationServiceConnectionManager.EventCallbacks eventCallbacks =
-      new FusedLocationServiceConnectionManager.EventCallbacks() {
-    @Override public void onConnect(Context context) {
-      FusedLocationProviderApiImpl.this.context = context;
-
-      Intent intent = new Intent(context, FusedLocationProviderService.class);
-      context.startService(intent);
+  @Override public void onConnect(Context context) {
+    this.context = context;
+    final Intent intent = new Intent(context, FusedLocationProviderService.class);
+    context.startService(intent);
+    context.bindService(intent, this, Context.BIND_AUTO_CREATE);
+  }
 
-      intent = new Intent(context, FusedLocationProviderService.class);
-      context.bindService(intent, serviceConnection, Context.BIND_AUTO_CREATE);
+  @Override public void onServiceConnected(IBinder binder) {
+    FusedLocationProviderService.FusedLocationProviderBinder fusedBinder =
+        (FusedLocationProviderService.FusedLocationProviderBinder) binder;
+    if (fusedBinder != null) {
+      service = fusedBinder.getService();
+      isBound = true;
     }
+  }
 
-    @Override public void onServiceConnected(IBinder binder) {
-      FusedLocationProviderService.FusedLocationProviderBinder fusedBinder =
-          (FusedLocationProviderService.FusedLocationProviderBinder) binder;
-      if (fusedBinder != null) {
-        service = fusedBinder.getService();
-      }
+  @Override public void onDisconnect() {
+    if (isBound) {
+      context.unbindService(this);
```

Comment:
I think we should set `isBound = false` here. `onServiceDisconnected` is not called when the client is unbound normally. According to the docs, `onServiceDisconnected` is only called when the process hosting the service is killed/crashed.